### PR TITLE
Fix bug in /msg

### PR DESCRIPTION
**Pull request by TheJetOU**
Reported here: https://www.smogon.com/forums/posts/8114587

### CODE REVIEW

File: server/chat.js [hunk 1]
```diff
@@ -386,7 +386,7 @@ class MessageContext {
 		if (commaIndex < 0) {
 			return [target.trim(), ''];
 		}
-		return [target.substr(0, commaIndex).trim(), target.substr(commaIndex + 1).trim()];
+		return [target.slice(0, commaIndex).trim(), target.slice(commaIndex + 1).trim()];
```
Comment (TheJetOU) [inline server/chat.js]: This is unnecessary for the fix, but I just thought it was clearer for future people reading the code.

File: server/chat-commands.js [hunk 2]
```diff
@@ -473,14 +473,15 @@ const commands = {
...
+		if (!target.includes(',')) {
+			this.errorReply("You forgot the comma.");
+			return this.parse('/help msg');
+		}
 		target = this.splitTarget(target);
 		let targetUser = this.targetUser;
 		if (this.targetUsername === '~') {
 			this.room = Rooms.global;
 			this.pmTarget = null;
-		} else if (!target) {
```
Comment (Zarel) [inline server/chat-commands.js]: This check still does need to be done, just with a different message, like "You can't leave your message empty."

Comment (TheJetOU) [reply server/chat-commands.js]: Isn't that handled here? https://github.com/Zarel/Pokemon-Showdown/blob/d6b4705870849d54543038321d3b5c898beea0b6/server/chat.js#L1162-L1165

Comment (Zarel) [reply server/chat-commands.js]: Good point. Yeah, that should be good.

**Pull request merged**